### ISSUE TITLE
Decorating Flask MethodView with multiple decorators results in "No jwt_data found in app context" error.

### ISSUE THREAD

**Issue by adyouri** [dusktreader/flask-praetorian#135]
Hi, thanks for creating this extension.

Using `MethodView` in the following manner results in an error:

```
class Protected(MethodView):
    decorators = [flask_praetorian.auth_required,
                  flask_praetorian.roles_required('admin')
                  ]
    def get(self):
        return flask.jsonify({'message': 'protected'})

app.add_url_rule('/protected_view/',
                 view_func=Protected.as_view('protected_view'))
```

When commenting out `flask_praetorian.roles_required('admin')` it works fine, but specifying both decorators results in the following error:

```
{
  "error": "PraetorianError", 
  "message": "No jwt_data found in app context.\nMake sure @auth_required decorator is specified *first* for route", 
  "status_code": 401
}
```
How exactly should I specify `@auth_required` first?

I initially faced this issue with Flask-RESTPlus, and since it relies on `MethodView`, I assumed it is probably the root of the issue.

Thank you!

**Comment by dusktreader** [comment]
@adyouri I just noticed this issue today. Sorry it slipped past! The issue here may be the ordering of the decorators. Try reversing them and see what happens. The problem, I think is, that it's trying to evaluate the roles_required first, but it doesn't have any jwt data because that's parsed out in the auth_required decorator. I believe I call out how the ordering is supposed to be used for normal decorators. When explicitly specifying them as a list class property for MethodView the order may need to be reversed. Please try that out and let me know. Perhaps I can document it as such.

One solution for this could possibly be to make the auth_required implicit when roles_required or roles_accepted is used

**Comment by dusktreader** [comment]
It's turning out to be trickier than I thought to have an implicit auth_required. I'm still working on it, though

**Comment by dusktreader** [comment]
@adyouri This has been addressed by pull-request #137 .

I made the @auth_required decorator optional when using @roles_required or @roles_accepted. The effect is that order of inclusion should not matter now, and you can just add the roles_required decorator by itself now.

**Comment by adyouri** [comment]
@dusktreader Thank you for your work! I tested it, and it works as expected with Flask-RESTPlus.
